frontend/types.asdl

```diff
@@ -8,6 +8,7 @@ module types {
               generate [integers]
 
   # Fifteen lexer modes for OSH
+  # TODO(GABE): should this be 16 now?
```

Comment:
I didn't really understand what this meant, but I counted 15 lines (not 15 tokens) under lex_mode. Not sure if we want `HereDoc` to be on the same line as DQ (or if that even matters)?